File: Satinalma Pro/Services/StokBelgeNoUretici.cs
using SatinalmaPro.Models;

namespace SatinalmaPro.Services;

public static class StokBelgeNoUretici
{
    public static string SonrakiGirisBelgeNo() => SonrakiBelgeNo("GR");

    public static string SonrakiCikisBelgeNo() => SonrakiBelgeNo("CK");

    private static string SonrakiBelgeNo(string kod)
    {
        var yil = DateTime.Now.Year;
        var onEk = $"{kod}-{yil}-";
        var sira = ModulVeriDeposu.StokHareketleri
            .Select(h => h.BelgeNo)
            .Where(b => b.StartsWith(onEk, StringComparison.OrdinalIgnoreCase))
            .Select(SiraNumarasiAl)
            .DefaultIfEmpty(0)
            .Max();

        return $"{onEk}{sira + 1:D3}";
    }

    private static int SiraNumarasiAl(string belgeNo)
    {
        var sonTire = belgeNo.LastIndexOf('-');
        if (sonTire < 0 || sonTire >= belgeNo.Length - 1)
            return 0;

        return int.TryParse(belgeNo[(sonTire + 1)..], out var no) ? no : 0;
    }
}

File: SatinalmaPro.Android/app/src/main/java/com/satinalmapro/android/core/StokBelgeNoUretici.kt
package com.satinalmapro.android.core

import com.satinalmapro.android.core.model.StokHareket
import java.util.Calendar

object StokBelgeNoUretici {
    fun sonrakiGirisBelgeNo(hareketler: List<StokHareket>): String = sonrakiBelgeNo("GR", hareketler)

    fun sonrakiCikisBelgeNo(hareketler: List<StokHareket>): String = sonrakiBelgeNo("CK", hareketler)

    private fun sonrakiBelgeNo(kod: String, hareketler: List<StokHareket>): String {
        val yil = Calendar.getInstance().get(Calendar.YEAR)
        val onEk = "$kod-$yil-"
        val sira = hareketler
            .map { it.belgeNo }
            .filter { it.startsWith(onEk, ignoreCase = true) }
            .mapNotNull { belge ->
                belge.substringAfterLast('-').toIntOrNull()
            }
            .maxOrNull() ?: 0
        return "$onEk${(sira + 1).toString().padStart(3, '0')}"
    }
}
